File: CustomQueries/server/aspnet/RevealSdk.Server/Reveal/UserContextProvider.cs
using Reveal.Sdk;

namespace RevealSdk.Server.Reveal;

public class UserContextProvider : IRVUserContextProvider
{
    public IRVUserContext GetUserContext(HttpContext aspnetContext)
    {
        //when using standard auth mechanisms, the userId can be obtained using aspnetContext.User.Identity.Name.
        var userIdentityName = aspnetContext.User.Identity?.Name;
        var userId = userIdentityName ?? "guest";

        //add the sales-person-id property 
        var props = new Dictionary<string, object> { { "sales-person-id", 279 } };

        return new RVUserContext(userId, props);
    }
}

File: CustomQueries/server/spring-boot-jersey/src/main/java/com/server/reveal/UserContextProvider.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.api.IRVUserContextProvider;
import com.infragistics.reveal.sdk.base.RVUserContext;

import java.util.HashMap;

public class UserContextProvider implements IRVUserContextProvider {
    @Override
    public IRVUserContext getUserContext() {
        // this can be used to store values coming from the request.
        var props = new HashMap<String, Object>();

        //add the sales-person-id property
        props.put("sales-person-id", 279);

        return new RVUserContext("user identifier", props);
    }
}
